### ISSUE TITLE
How can I configure RDP settings for an individual connection?

### ISSUE THREAD

**Issue by jdx-john** [mRemoteNG/mRemoteNG#2357]
Is there a way to access the normal RDP settings somehow if I wish to customise a connection? e.g. downgrade the experience for a remote machine on a very limited network?

![image](https://user-images.githubusercontent.com/1654627/219622661-bc0db6bc-adc4-4907-8130-f4acccee1880.png)

I get the impression(?) that mRemoteNG is actually using RDP behind the scenes (?) so is it actually saving an RDP connection file for each connection somewhere that I can open directly and edit?

**Comment by simonai1254** [comment]
You are correct that mRemoteNG leverages the Microsoft RDP framework, but the connections are not stored as a .rdp connection file. 

The preferences you are looking for can be found in the connection settings in the "Appearance" section: 
![image](https://user-images.githubusercontent.com/33432894/219665849-6ced1b90-a5b5-46bd-b3ac-171c1c1dd517.png)

Fit to Panel automatically chooses the most ideal resolution based on the window size and prevents that you might need to scroll when the RDP resolution is higher than the actual screen you are using.

Just for your interest: Depending on how you use RDP, the resolution does not matter that much regarding bandwith usage as only the changed elements are transmitted: https://learn.microsoft.com/en-us/azure/virtual-desktop/rdp-bandwidth

**Comment by jdx-john** [comment]
Thanks @simonai1254 , forgive my probably foolishness but I don't see an Appearance section at all:

![image](https://user-images.githubusercontent.com/1654627/219671653-0fa988d9-4742-40d6-8da9-4ebfc295cec5.png)


On scaling generally I see the "Smart Size" option but the issue is on my target systems, I need the desktop to be quite large or the apps simply don't work right - I connect and everything is all cramped. When we connect using raw RDP sometimes the RDP window even has scroll-bars. What I want is for the remote desktop to be something like 1600x1200 and scaled to fit the mREmoteNG window. I am fairly sure it can do it but I'm not sure how. I know that partly this is dependent on the client's RDP version too, in terms of what is supported (all our machines are Windows server, often older versions)

v1.76.20 btw (I think that's current as of this week)

**Comment by simonai1254** [comment]
mRemoteNG offerst Resolutions up to 4K to choose in the window, so this shouldn't be an issue. 

Something indeed looks off, you should have a lot more config items displayed. At least you should be able to switch between SSH and RDP Connection type, which is not displayed. Maybe try "View">"Reset Layout" and see if it helps.

Otherwise I would suggest the following two steps:

- Try 1.77.2-nb (the current nightly build that allows you to save configuration changes), which can be downloaded [in the releases](https://github.com/mRemoteNG/mRemoteNG/releases)
- Try a complete new configuration file for mRemoteNG and add a new connection

May I ask how you added the connection? Was it out of a "quick connect"?

Hope this helps

**Comment by jdx-john** [comment]
I was given an XML file with all the machines our project uses. Is it possible this is from an old version that has some incompatibilities perhaps, likely this file has been passed around for ages? If so would there be any way to port it... there are dozens of entries and doing it by hand would be somewhat dull.
I can test creating a brand new connection file and recreating one target connection from scratch.

**Comment by simonai1254** [comment]
I'm not sure how the connection files behave when upgrading.
I would first try to rebuild one single connection and test that. If this fixes the issue, then we can look into how to migrate the current settings into a new configuration file. Generally my first try would be to export the connections into a new file, and then import these settings into a new clean configuration file. If this doesn't work it might be easiest to directly edit the XML, but at that point one could rebuild the connections as well.

**Comment by Kvarkas** [comment]
As far as I am aware we dont change schema so connection file should be compatible for versions 1.77.x

**Comment by Kvarkas** [comment]
@jdx-john hello, any luck with convertion? Are you still need support on such?

**Comment by Kvarkas** [comment]
closing as seems not an issue any more
